Changes

```diff
@@ -205,6 +205,12 @@ Working version
 - GPR#1886: move the Location.absname reference to Clflags.absname
   (Armaël Guéneau, review by Jérémie Dimino)
 
+- GPR#374: the prototype of `Misc.try_finally` has changed by using optional
+  arguments. Cleanup in exceptional case have been added (`~exceptionally`). It
+  uses `Printexc.raise_with_backtrace` for preserving backtraces.
+  (François Bobot, Gabriel Scherer, and Nicolás Ojeda Bär, review by Gabriel
+  Scherer)
```

Comment:
Reading this change entry again, I believe that it doesn't reflect the content of the PR anymore. Here would be a suggestion for a replacement:

```
- GPR#374: use Misc.try_finally for resource cleanup in the compiler codebase. This should
  fix the problem of catch-and-reraise `try .. with` blocks destroying backtrace information.
  (credits)
```

(Semi-related question: does the use of `protect_refs` drops backtraces? Its implementation is not using `try_finally` itself, it probably should, but it may be that the "detect raises that are re-raises" logic suffice in this particular case, as the exception-cleanup work does not involved raising exceptions.)

Reply:
Thanks, amended as suggested. And yes, I believe `Misc.protect_refs` preserves backtraces.